### ISSUE TITLE
Lock Drawer

### ISSUE THREAD

**Issue by vineetkumar198715** [react-community/react-navigation#921]
I am using DrawerNavigation which is working fine. But I want to disable/lock the drawer on some specific screens. How can I do that?

**Comment by dominicrj23** [comment]
You can create your own `DrawerView`, create `drawerRouter` with your routeConfig and then use it the custom Navigator helpers
```
createNavigationContainer(createNavigator(drawerRouter)(DrawerView))
```

**Comment by vineetkumar198715** [comment]
Creating the Drawer is not the issue. I have been able to create the Drawer and the navigation between screens is also working fine. The only issue that I  am facing is I want to disable the Drawer or do not want to show the Drawer on some specific pages. I have seen the "drawerLockMode" option but that is available only for drawerLayoutAndroid API. I want to do this on DrawerNavigator that is available on reactnavigation.org site.

**Comment by ericvicenti** [comment]
Can you try this PR and let us know if it works for your needs?

https://github.com/react-community/react-navigation/pull/793

**Comment by RWOverdijk** [comment]
Locking the drawer on nested stack navigators is an excellent use-case for this. The PR @ericvicenti points to would work, but means you have to define it either on all nested routers, or disable it by default and enable it on root screens.

**Comment by nonameolsson** [comment]
This one could be closed, right? #1377

**Comment by vonovak** [comment]
this was resolved in #1377
